src/NHibernate/Multi/LinqBatchItem.cs

```diff
@@ -9,8 +10,15 @@
 
 namespace NHibernate.Multi
 {
+	interface ILinqBatchItem
+	{
+		List<TResult> GetTypedResults<TResult>();
+	}
+
 	public static class LinqBatchItem
 	{
+		internal static ConcurrentDictionary<System.Type, Func<ILinqBatchItem, IList>> GetResultsForTypeDic = new ConcurrentDictionary<System.Type, Func<ILinqBatchItem, IList>>();
```

Comment:
Let's generate compiled delegate returning typed results and cache it.

It's OK to have static cache as it will contain only few items (and for now it's only for nullable types)